### ISSUE TITLE
Scalar indexing can occur on `show` for arrays of DiskArray

### ISSUE THREAD

**Issue by rafaqz** [meggart/DiskArrays.jl#85]
This doesn't occur when the DiskArray is shown on its own. This may be from alternate methods of `show` being called for an array of array. We may need to define show without a `MIME` type?

**Comment by rafaqz** [comment]
Ahh no, this is when the generic method is used on a wrapper array type.

It's annoying that we have to handle this in all wrappers by explicitly forwarding `show`.

**Comment by meggart** [comment]
Yes, I find this annoying as well indeed. I have no idea how to solve this though as long as there are no Array traits that are accepted by `show` for Abstract Arrays in Julia Base. Do you happen to know how the gpu people are solving this when dealing e.g. with CuArrays? They should have pretty slow random access as well (but probably not as slow as compressed arrays on disk/in the cloud).

**Comment by rafaqz** [comment]
Yeah we should look at what CuArrays do. Since they have `allowscalar` too they would have to avoid that error somehow on show?

Thats how I found this, setting `DiskArrays.allowscalar(false)` means `show` breaks if you have e.g. a vector of wrapped disk arrays.

I'll handle it in Rasters.jl by adding `show` without MIME and checking if the parent array is disk based. But automatically forwarding a trait to parent would be best long term, somehow or other.
